--- origsrc/lives-0.9.9.4/configure.in	2008-12-03 10:17:07.000000000 -0600
+++ src/lives-0.9.9.4/configure.in	2008-12-29 00:20:24.096875000 -0600
@@ -42,11 +42,15 @@
 exit
 ])
 
+IS_CYGWIN="false"
 IS_DARWIN="false"
 IS_IRIX="false"
 IS_LINUX_GNU="false"
 
 case "{$target_os}" in
+ *cygwin*)
+   IS_CYGWIN="true"
+  ;;
  *darwin*)
    IS_DARWIN="true"
   ;;
@@ -60,6 +64,7 @@
   ;;
 esac
 
+AM_CONDITIONAL(IS_CYGWIN,$IS_CYGWIN)
 AM_CONDITIONAL(IS_DARWIN,$IS_DARWIN)
 AM_CONDITIONAL(IS_IRIX,$IS_IRIX)
 AM_CONDITIONAL(IS_LINUX_GNU,$IS_LINUX_GNU)
--- origsrc/lives-0.9.9.4/libweed/Makefile.am	2008-09-06 16:31:24.000000000 -0500
+++ src/lives-0.9.9.4/libweed/Makefile.am	2008-12-29 00:20:24.112500000 -0600
@@ -11,8 +11,8 @@
 
 extra_cflags = -fPIC -DPIC
 endif
 
-lib_LTLIBRARIES = libweed.la libweed_slice.la libweed-utils.la libweed-plugin-utils.la
+noinst_LTLIBRARIES = libweed.la libweed_slice.la libweed-utils.la libweed-plugin-utils.la
 
 libweed_la_SOURCES = weed.c
 
--- origsrc/lives-0.9.9.4/lives-plugins/plugins/decoders/Makefile.am	2008-11-03 19:48:40.000000000 -0600
+++ src/lives-0.9.9.4/lives-plugins/plugins/decoders/Makefile.am	2008-12-29 00:26:48.128125000 -0600
@@ -8,9 +8,14 @@
 shared_ldflags = -O3 -dynamiclib -module -avoid-version --tag=disable-static
 shared_cflags = -fno-common
 else
+if IS_CYGWIN
+shared_ldflags = -O3 -shared -module -avoid-version -no-undefined --tag=disabled-static
+shared_cflags = -O3 -DPIC
+else
 shared_ldflags = -O3 -shared -fPIC -DPIC -module -avoid-version --tag=disable-static
 shared_cflags = -O3 -fPIC -DPIC
 endif
+endif
 
 
 if HAVE_LIBDV
--- origsrc/lives-0.9.9.4/lives-plugins/plugins/playback/video/Makefile.am	2008-11-12 05:05:19.000000000 -0600
+++ src/lives-0.9.9.4/lives-plugins/plugins/playback/video/Makefile.am	2008-12-29 00:20:24.128125000 -0600
@@ -8,9 +8,14 @@
 shared_ldflags = -O3 -dynamiclib -module -avoid-version --tag=disable-static
 shared_cflags = -fno-common
 else
+if IS_CYGWIN
+shared_ldflags = -O3 -shared -module -avoid-version -no-undefined --tag=disable-static
+shared_cflags = -O3 -DPIC
+else
 shared_ldflags = -O3 -shared -fPIC -DPIC -module -avoid-version --tag=disable-static
 shared_cflags = -O3 -fPIC -DPIC
 endif
+endif
 
 if HAVE_SDL
 SDL_libs = SDL.la
--- origsrc/lives-0.9.9.4/lives-plugins/plugins/playback/video/lives2lives_stream.c	2008-07-29 02:13:33.000000000 -0500
+++ src/lives-0.9.9.4/lives-plugins/plugins/playback/video/lives2lives_stream.c	2008-12-29 00:20:24.143750000 -0600
@@ -3,6 +3,7 @@
 // released under the GNU GPL 3 or later
 // see file COPYING or www.gnu.org for details
 
+#include <netinet/in.h>
 #include "../../../../libweed/weed.h"
 #include "../../../../libweed/weed-effects.h"
 
--- origsrc/lives-0.9.9.4/lives-plugins/weed-plugins/Makefile.am	2008-11-04 05:52:26.000000000 -0600
+++ src/lives-0.9.9.4/lives-plugins/weed-plugins/Makefile.am	2008-12-29 00:20:24.159375000 -0600
@@ -11,10 +11,14 @@
 
 shared_cflags = -fno-common
 else
+if IS_CYGWIN
+shared_ldflags = -shared -module -avoid-version -no-undefined --tag=disable-static
+else
 shared_ldflags = -shared -fPIC -DPIC -Wl,-Bsymbolic -module -avoid-version --tag=disable-static
 
 shared_cflags = -fPIC -DPIC
 endif
+endif
 
 AM_LDFLAGS = $(shared_ldflags) -O3 -L/usr/lib64 ../../libweed/libweed-utils.la ../../libweed/libweed-plugin-utils.la
 AM_CFLAGS = $(shared_cflags) -O3 -ffast-math
--- origsrc/lives-0.9.9.4/lives-plugins/weed-plugins/gdk/Makefile.am	2008-11-03 19:48:40.000000000 -0600
+++ src/lives-0.9.9.4/lives-plugins/weed-plugins/gdk/Makefile.am	2008-12-29 00:20:24.159375000 -0600
@@ -8,9 +8,13 @@
 shared_ldflags = -dynamiclib -Wl,-Bsymbolic -module -avoid-version --tag=disable-static
 shared_cflags = -fno-common
 else
+if IS_CYGWIN
+shared_ldflags = -shared -module -avoid-version -no-undefined --tag=disable-static
+else
 shared_ldflags = -shared -fPIC -DPIC -Wl,-Bsymbolic -module -avoid-version --tag=disable-static
 shared_cflags = -fPIC -DPIC
 endif
+endif
 
 AM_LDFLAGS = $(shared_ldflags) @GTK_LIBS@ -O3 -L/usr/lib64  ../../../libweed/libweed-utils.la ../../../libweed/libweed-plugin-utils.la
 AM_CFLAGS = $(shared_cflags) -I$(srcdir)/.. @GTK_CFLAGS@ -O3 -ffast-math
--- origsrc/lives-0.9.9.4/src/htmsocket.c	2008-07-11 01:02:34.000000000 -0500
+++ src/lives-0.9.9.4/src/htmsocket.c	2008-12-29 00:20:24.175000000 -0600
@@ -11,6 +11,7 @@
 #include <errno.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
+#include <netinet/in.h>
 
 #include "main.h"
 #include "support.h"
--- origsrc/lives-0.9.9.4/src/jack.h	2008-12-02 21:40:24.000000000 -0600
+++ src/lives-0.9.9.4/src/jack.h	2008-12-29 00:20:24.190625000 -0600
@@ -11,6 +11,7 @@
 
 /////////////////////////////////////////////////////////////////
 // Transport
+#include <sys/time.h>
 #include <jack/jack.h>
 #include <jack/transport.h>
 
--- origsrc/lives-0.9.9.4/src/plugins.c	2008-12-03 08:07:22.000000000 -0600
+++ src/lives-0.9.9.4/src/plugins.c	2008-12-29 00:20:24.206250000 -0600
@@ -931,7 +931,7 @@
 
   // TODO - if using, get fixed_fps,fwidth,fheight,palette,argc and argv from a file
 
-  gchar *plugname=g_strdup_printf ("%s%s%s/%s.so",prefs->lib_dir,PLUGIN_EXEC_DIR,PLUGIN_VID_PLAYBACK,name);
+  gchar *plugname=g_strdup_printf ("%s%s%s/%s.%s",prefs->lib_dir,PLUGIN_EXEC_DIR,PLUGIN_VID_PLAYBACK,name,G_MODULE_SUFFIX);
   void *handle=dlopen(plugname,RTLD_LAZY);
   gboolean OK=TRUE;
   gchar *msg,*tmp;
@@ -1583,7 +1583,7 @@
   g_free(decplugdir);
 
   while (decoder_plugins!=NULL) {
-    gchar *decplugname=g_strdup_printf("%s.so",(gchar *)decoder_plugins->data);
+    gchar *decplugname=g_strdup_printf("%s.%s",(gchar *)decoder_plugins->data,G_MODULE_SUFFIX);
     dplug=open_decoder_plugin(decplugname,sfile);
 
     //#define DEBUG_DECPLUG
