--- origsrc/lives-1.0.6/libweed/Makefile.am	2009-11-18 02:56:44.487728700 -0600
+++ src/lives-1.0.6/libweed/Makefile.am	2009-11-19 23:14:21.495695600 -0600
@@ -35,9 +35,7 @@ libweed_utils_la_SOURCES = weed-utils.c 
 
 libweed_plugin_utils_la_SOURCES = weed-plugin-utils.c
 
-libweedinclude_HEADERS = weed.h weed-host.h weed-plugin.h weed-utils.h weed-effects.h weed-events.h weed-plugin-utils.h
-
-libweedincludedir = $(includedir)/weed
+noinst_HEADERS = weed.h weed-host.h weed-plugin.h weed-utils.h weed-effects.h weed-events.h weed-plugin-utils.h
 
 libweed_ladir = $(libdir)/weed
 
