--- origsrc/lives-1.6.1/libweed/Makefile.am	2012-02-09 17:30:46.792231100 -0600
+++ src/lives-1.6.1/libweed/Makefile.am	2012-02-09 22:48:40.577003300 -0600
@@ -13,8 +13,6 @@
 
 install-data-local:
 	make docs-doxygen
-	mkdir -p "$(DESTDIR)$(libdir)/pkgconfig/"
-	cp -f libweed*.pc "$(DESTDIR)$(libdir)/pkgconfig/"
 
 if HAVE_DOXYGEN
 docs-doxygen:
@@ -40,7 +38,7 @@ endif
 
 EXTRA_DIST = libweed.doxygen libweed.pc libweed-plugin-utils.pc libweed_slice.pc libweed-utils.pc libweed-compat.pc
 
-lib_LTLIBRARIES = libweed.la libweed_slice.la libweed-utils.la libweed-plugin-utils.la
+noinst_LTLIBRARIES = libweed.la libweed_slice.la libweed-utils.la libweed-plugin-utils.la
 
 libweed_la_SOURCES = weed.c
 
@@ -50,7 +48,7 @@ libweed_utils_la_SOURCES = weed-utils.c
 
 libweed_plugin_utils_la_SOURCES = weed-plugin-utils.c
 
-libweedinclude_HEADERS = weed.h weed-host.h weed-plugin.h weed-utils.h weed-events.h weed-palettes.h weed-effects.h \
+noinst_HEADERS = weed.h weed-host.h weed-plugin.h weed-utils.h weed-events.h weed-palettes.h weed-effects.h \
 	weed-plugin-utils.h weed-compat.h
 
 libweedincludedir = $(includedir)/weed
